044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,342 @@
+
+## Rejected Alternatives
+
+### Using an Authorizer Plugin
+
+Using an [authorizer plugin](https://kafka.apache.org/33/javadoc/org/apache/kafka/server/authorizer/Authorizer.html) 
+to deny operations is another alternative. It would allow finer-grained control but would potentially be surprising 
+to kafka clients to see unexpected authorization denials. Also, the API would need careful attention to performance 
+as it's called per-operation. 
+
+Enabling it to chain with another user-configured authorizer is another complexity. The plugin would have to be made 
+aware of the delegate class, instantiate it and drive its lifecycle.
+
+### Using a kafka topic to distribute metrics. 
+
+See the original proposal PR [#51](https://github.com/strimzi/proposals/pull/51). We considered using a kafka topic to
+push volume usage metrics out to all the brokers. KIP-827 made this redundant.
+
+### Using JMX metrics
+
+Using JMX metrics directly would require a web of connections between brokers and the exposing of the JMX port to the
+rest of the cluster. Using JMX is also problematic for tracking state across restarts of brokers as each broker would
+lose state across restarts and thus lose track of any broker which is temporarily offline.
+
+### External metrics store
+
+Would require the following:
+
+- The quota plugin understands the API of the external metrics system
+- A metrics system endpoint exposed to the broker for consuming metrics
+- A predictable and consistent naming convention
+
+It would also make the deployment of an external metrics store a requirement for the kafka-static-quota-plugin to function.
+
+### KIP-73
+
+KIP-73 is designed to protect client performance while cluster re-balancing exercises are taking place by limiting the
+bandwidth available to the replication traffic. This is not suitable for use in preventing out of storage issues as the
+bandwidth limit is configured as part of the partition re-assignment operation. As it applies a bandwidth limit it is
+configured in `units per second` which is problematic for the quota plugin to determine a sensible value for as it
+should really be related to the expected rate at which data is purged from the tail of the partitions on the volume in
+question. KIP-73 bandwidth limits are only applied to a specific set of `partition` & `replica` pairs which would
+require the ability for the plugin to resolve the required pairs.
+
+## Affected projects
+* strimzi/kafka-quota-plugin
+* strimzi/kafka-cluster-operator
+
+## Compatibility
+Backwards compatibility would be maintained while kafka 3.2 is a supported version. We want to support users of older 
+kafka brokers as the local volume source can protect users in some scenarios (homogeneous disks with well-balanced data).
+
+Attempting to use a `cluster` volume source with a kafka older than 3.3.0 will prevent the broker from starting up
+and emit some sane log indicating the broker version is incompatible.
```

Comment:
Strimzi can pull different quotas plugin versions into different Kafka versions. So we will not pull the new version into Kafka 3.2. But also, Kafka 3.4 will be out soon, so we will likely drop Kafka 3.2 before this is implemented and released.